File: apple/packages/EnteComponents/Sources/EnteComponents/Settings/SettingsPage.swift
import SwiftUI

public struct SettingsPage<Content: View>: View {
    private let title: String
    private let identifier: String?
    private let backLabel: String
    private let back: () -> Void
    private let content: Content

    public init(
        _ title: String,
        identifier: String? = nil,
        backLabel: String,
        back: @escaping () -> Void,
        @ViewBuilder content: () -> Content,
    ) {
        self.title = title
        self.identifier = identifier
        self.backLabel = backLabel
        self.back = back
        self.content = content()
    }

    public var body: some View {
        CollapsingPage(title, identifier: identifier, backLabel: backLabel, back: back) {
            VStack(alignment: .leading, spacing: EnteSpacing.sm) { content }
        }
    }
}

File: android/packages/components/src/main/kotlin/io/ente/components/settings/SettingsPage.kt
package io.ente.components.settings

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.safeDrawingPadding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import io.ente.components.CollapsingPage
import io.ente.components.EnteSpacing

@Composable
public fun SettingsPage(
    title: String,
    backLabel: String,
    back: () -> Unit,
    modifier: Modifier = Modifier,
    identifier: String? = null,
    content: @Composable ColumnScope.() -> Unit,
) {
    CollapsingPage(title, backLabel, back, modifier.safeDrawingPadding(), identifier) {
        item {
            Column(verticalArrangement = Arrangement.spacedBy(EnteSpacing.sm), content = content)
        }
    }
}
